src/Core/System/Snippet/Struct/SnippetValidationStruct.php

```diff
@@ -0,0 +1,32 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\Snippet\Struct;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Struct\Struct;
+
+/**
+ * @phpstan-type MissingSnippets array<string, array<string, array{
+ *      path: string,
```

Comment:
why stopping at the first level? :grin: 

object shape array annotations are the first sign of "hey, this should be a class instead" :wink:

Reply:
You really do not think it is overengineering that data went from Array to Struct in a Struct?

Reply:
well on another place you said, you had difficulties to describe what the arrays contain. with classes this is well described and also typed. for sure, it might sound easier to just throw everything in arrays, but they are worse in every regards in comparison to objects. maybe worth a read :slightly_smiling_face: -> 
https://peakd.com/php/@crell/php-use-associative-arrays-basically-never
https://presentations.garfieldtech.com/slides-never-use-arrays/phpday2021/